app/containers/EditStudentContainer.js

```diff
@@ -0,0 +1,126 @@
+import React, {Component} from 'react';
+import store from '../store';
+import EditStudent from '../components/EditStudent';
+import axios from 'axios';
+import {
+  receiveStudents,
+  updateStudent,
+  setFirstStudentName,
+  setLastStudentName,
+  setStudentEmail,
+  setStudentCampus,
+  getStudentById,
+} from '../action-creators/students';
+import {
+  receiveCampuses,
+} from '../action-creators/campuses';
+
+const initialState = {
+  studentFirstName: '',
+  studentLastName: '',
+  studentEmail: '',
+  studentCampus: '',
+};
+
+class EditStudentContainer extends Component {
+  constructor() {
+    super();
+    this.state = store.getState();
+    this.handleFirstNameChange = this.handleFirstNameChange.bind(this);
+    this.handleLastNameChange = this.handleLastNameChange.bind(this);
+    this.handleEmailChange = this.handleEmailChange.bind(this);
+    this.handleCampusChange = this.handleCampusChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    this.unsubscribe = store.subscribe(() =>
+      this.setState(store.getState())
+    )
+    Promise.all([
```

Comment:
There should be only one place in the application that requests this data and inserts it into the `redux` store. All other components should then read from/subscribe to the store.